src/Workspaces/Core/Portable/Rename/ConflictEngine/ConflictResolver.Session.cs

```diff
@@ -746,11 +753,34 @@ private async Task AddDocumentsWithPotentialConflicts(IEnumerable<Document> docu
             {
                 try
                 {
+                    var annotatedSolution = originalSolution;
+
+                    // Add the RenameSymbolAnnotation to the declarations
+                    foreach (var syntaxReference in _renameLocationSet.Symbol.DeclaringSyntaxReferences)
+                    {
+                        var document = annotatedSolution.GetDocument(syntaxReference.SyntaxTree);
+                        if (document is null)
+                        {
+                            continue;
+                        }
+
+                        var node = await syntaxReference.GetSyntaxAsync().ConfigureAwait(false);
+                        var root = await document.GetSyntaxRootAsync().ConfigureAwait(false);
+
+                        var newRoot = root.ReplaceNode(
+                            node,
+                            node.WithoutAnnotations(RenameSymbolAnnotation.RenameSymbolKind)
+                                .WithAdditionalAnnotations(RenameSymbolAnnotation.Create(_renameLocationSet.Symbol)));
+
+                        var annotatedDocument = document.WithSyntaxRoot(newRoot);
+                        annotatedSolution = annotatedDocument.Project.Solution;
+                    }
+
                     foreach (var documentId in documentIdsToRename.ToList())
                     {
                         _cancellationToken.ThrowIfCancellationRequested();
 
-                        var document = originalSolution.GetDocument(documentId);
+                        var document = annotatedSolution.GetDocument(documentId);
```

Comment:
Should "originalSolution" not be used after this point? Rather than setting that trap, should we just overwrite "originalSolution" and rename the local accordingly?

Reply:
That's fine with me. Renamed to just `solution`, but open to suggestions.